### ISSUE TITLE
bug in observation_metadata_correlation.py

### ISSUE THREAD

**Issue by wdwvt1** [biocore/qiime#2009]
`observation_metadata_correlation.py` has a show-stopping bug that causes the script to return incorrect values for any set of mapping file, biom file where the order of the samples returned by iterating over the dictionary representation of the mapping file doesn't end up being the same as the order of the samples in the biom table (I would guess these almost never line up).

Line 202 of `scripts/observation_metadata_correlation.py` is where the error is. It is simple:

```python
# sort the biom table so that feature values are retrieved in the same
# order as the metadata in the samples they correspond to
bt.sort(sort_f = lambda _: samples_to_correlate, axis='sample')
```

`.sort` doesn't work in place, so the sorted biom table is not returned. Simply changing this to return a new biom table object solves the problem. 

See [here](https://groups.google.com/forum/#!topic/qiime-forum/AljnRFdDXsU) for the user first reporting this issue.

**Comment by wdwvt1** [comment]
The reason that QIIME was not catching this bug is because the error was in the script rather than the library file. A refactor of this script would be nice to move the offending functionality to library code.

**Comment by jairideout** [comment]
Fixed in #2027.
